### PR TITLE
Change Protocol  to use an array for methodSelectors directly

### PR DESCRIPTION
**Pull request by MarcusDenker**
- 40K less IdentitySets alocated
- all the arrays are much smaller

I changed the API slghtly to be more sane and return "self", not the added element or the collection changed.

### CODE REVIEW

**Comment by MarcusDenker** [comment]
very odd crash on full image load

**Comment by jecisc** [comment]
A PR of Esteban also had a segmentation fault while loading the packages. Maybe there is a bug in the VM?

**Comment by jecisc** [comment]
See: https://ci.inria.fr/pharo-ci-jenkins2/blue/organizations/jenkins/Test%20pending%20pull%20request%20and%20branch%20Pipeline/detail/PR-13412/2/pipeline/25

<blockquote><div><strong><a href="https://ci.inria.fr/pharo-ci-jenkins2/blue/organizations/jenkins/Test%20pending%20pull%20request%20and%20branch%20Pipeline/detail/PR-13412/2/pipeline/25">Jenkins</a></strong></div></blockquote>

File: src/Kernel/Protocol.class.st [hunk 1]
```diff
@@ -59,7 +61,7 @@ Protocol >> initialize [
 
 	super initialize.
 
-	methodSelectors := IdentitySet new.
+	methodSelectors := Array empty.
```
Comment (jecisc) [inline src/Kernel/Protocol.class.st]:
```suggestion
	self removeAllMethodSelectors.
```

Maybe we could rename it #resetMethodSelectors instead later

**Review by jecisc — APPROVED** [review]

**Comment by MarcusDenker** [comment]
I got the same crash on #13415

**Review by privat — APPROVED** [review]

**Comment by privat** [comment]
I got the same CI issue. I opened #13419

**Comment by jecisc** [comment]
testHaltOnCount – MacOSX64.Debugging.Utils.Tests.HaltTest

It's the second time I see this random failure this week :(

**Pull request merged**